### ISSUE TITLE
Is it necessary to monkey-patch?

### ISSUE THREAD

**Issue by michael-markl** [jeff-hykin/json_fix#2]
Hi @jeff-hykin !

First, thanks for your package! I've used it a lot in the past.

After your last update, my code somehow broke (when using json.dump instead of json.dumps).
After reading some docs, I realized that your desired result could possibly be achieved a lot easier.
The functions `json.dump` and `json.dumps` allow to pass an argument called `default` (See [here](https://github.com/python/cpython/blob/main/Lib/json/__init__.py#L153-L154)). This argument is in turn passed to the (original) JSONEncoder class (see [here](https://github.com/python/cpython/blob/12b5a3c5f50749c8b06be781c47872262c658b45/Lib/json/encoder.py#L97-L100)). The docs explicitly state that this function can be used to make the (original) JSONEncoder recognize other (types of) objects.
Hence, in my case, the desired result can be achieved simply using: 

```python
def json_default(obj: Any):
    if hasattr(obj.__class__, "__json__"):
        json_method = getattr(obj.__class__, "__json__")
        if callable(json_method):
            return json_method(obj)
    raise TypeError()

json.dump(my_obj, file, default=json_default)
```

It would be cool, if you could react to this and maybe add a clarification on the difference between your approach and my approach above to your README.md.

Thank you!
Best,
Michael

**Edit:** I just saw that your package is mostly concerned about *external code* doing the json encoding. Then, of course, if there's no way out, monkey patching might be the only viable option. However, maybe you can add the above approach to the README.md for users (like me) who happen to control the json encoding themselves, but who want to be able to adjust the serialization of objects using the `__json__` function.

**Comment by jeff-hykin** [comment]
Hello @michael-markl ! Glad to hear the package is being used.

> After your last update, my code somehow broke (when using json.dump instead of json.dumps).

Oh thats terrible, I wanted to keep this library extremely stable. I believe I was able to reproduce your issue and published `0.5.1` to fix it just now. I'm shocked I didn't find this sooner with how many projects I tested this on.


Sadly monkey patching is the only way in this case :/ I dislike it as much as you do, for exactly this kind of reason. The idea was to treat this module as a central patch; so as more people use it the more tested/stable it becomes. Thanks for reporting the issue, I'm sure others have run into this without me noticing.
